src/Ng/Catalog2Registration.cs

```diff
@@ -91,7 +96,68 @@ public void Run(string[] args, CancellationToken cancellationToken)
 
             Trace.TraceInformation("CONFIG source: \"{0}\" storage: \"{1}\" interval: {2} seconds", source, storageFactory, interval);
 
-            Loop(source, storageFactory, contentBaseAddress, unlistShouldDelete, verbose, interval, cancellationToken).Wait();
+            if (compressedStorageFactory != null)
+            {
+                Loop(source, new AggregateStorageFactory(storageFactory, new [] { compressedStorageFactory }, SecondaryStorageContentInterceptor), contentBaseAddress, unlistShouldDelete, verbose, interval, cancellationToken).Wait();
+            }
+            else
+            {
+                Loop(source, storageFactory, contentBaseAddress, unlistShouldDelete, verbose, interval, cancellationToken).Wait();
+            }
+        }
+
+        private StorageContent SecondaryStorageContentInterceptor(Uri primaryResourceUri, Uri secondaryResourceUri, StorageContent content)
+        {
+            var jTokenStorageContent = content as JTokenStorageContent;
+            if (jTokenStorageContent != null)
+            {
+                var token = jTokenStorageContent.Content.DeepClone();
+
+                if (primaryResourceUri.ToString().EndsWith("/index.json"))
+                {
+                    InterceptRegistrationIndex(primaryResourceUri, secondaryResourceUri, token);
+                }
+                else
+                {
+                    InterceptRegistrationPackage(primaryResourceUri, secondaryResourceUri, token);
+                }
+
+                return new JTokenStorageContent(token, content.ContentType, content.CacheControl);
+            }
+
+            return content;
+        }
+
+        private void InterceptRegistrationPackage(Uri primaryResourceUri, Uri secondaryResourceUri, JToken content)
+        {
+            var tokens = content.SelectTokens("@id")
+                .Union(content.SelectTokens("registration"))
+                .ToArray();
+
+            for (int i = 0; i < tokens.Length; i++)
+            {
+                tokens[i].Replace(
+                    ((string)tokens[i]).Replace(primaryResourceUri.ToString(), secondaryResourceUri.ToString()));
+            }
+        }
+
+        private void InterceptRegistrationIndex(Uri primaryResourceUri, Uri secondaryResourceUri, JToken content)
+        {
+            var tokens = content.SelectTokens("@id")
+                .Union(content.SelectTokens("items[*].@id"))
+                .Union(content.SelectTokens("items[*].catalogEntry.@id"))
+                .Union(content.SelectTokens("items[*].registration"))
+                .Union(content.SelectTokens("items[*].parent"))
+                .Union(content.SelectTokens("items.items[*].@id"))
```

Comment:
wouldn't this be items[*].items[*].@id ? in other words wouldn't the first items also be an array reference?